### PR TITLE
Add quicktip repeated queries

### PR DESCRIPTION
**Pull request by texnixe**

### CODE REVIEW

**Comment by distantnative** [comment]
@texnixe have we just done a terrible job at (not) reviewing your PR or is anything else blocking this?

**Comment by texnixe** [comment]
It's just that noone has been interested in this... so I was on the verge of putting in into the recipe graveyard.

**Comment by distantnative** [comment]
I don't think the inactivity relates to no interest - more with inability to shift focus, like with a lot of PRs not eh core too.
I could have a look on it the next days if you think it's still good :)

**Comment by texnixe** [comment]
I will take a look at it again next week and then decide what to do with it.

**Comment by lukasbestle** [comment]
@texnixe To be honest, I'm often not quite sure if your recipe PRs are ready for review yet, so I only pick them up if I have been requested for review via the GitHub UI

**Comment by texnixe** [comment]
@lukasbestle Yes, I understand and promise to explicitly add reviewers in the future. I know I can count on you!

File: content/1_docs/2_cookbook/3_templating/0_keep_your_code_dry/recipe.txt [hunk 1]
```diff
@@ -0,0 +1,133 @@
+Title: Quicktip: Don't repeat the same query multiple times
+
+----
+
+Published: 2021-01-09
+
+----
+
+Description: Don't repeat yourself is a coding paradigm that when not heeded can seriously harm performance.
```
Comment (lukasbestle) [inline content/1_docs/2_cookbook/3_templating/0_keep_your_code_dry/recipe.txt]:
```suggestion
Description: "Don't repeat yourself" (DRY) is a coding paradigm that when not heeded can seriously harm performance.
```

File: content/1_docs/2_cookbook/3_templating/0_keep_your_code_dry/recipe.txt [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+Furthermore, this often results in unexpected behaviour. Consider this example:
+
+```php "/site/controllers/blog.php"
+<?php
+
+return function($page) {
+  $articles = $page->articles()->listed();
+  if($tag = param('tag')) {
+    $articles = $articles->filterBy('tags', $tag, ',');
```
Comment (lukasbestle) [inline content/1_docs/2_cookbook/3_templating/0_keep_your_code_dry/recipe.txt]:
```suggestion
    $articles = $articles->filter('tags', $tag, ',');
```

File: content/1_docs/2_cookbook/3_templating/0_keep_your_code_dry/recipe.txt [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  }
+  return [
+    'articles' => $articles,
+  ];
+};
+```
+
+If we now use the same template code as in the example above, we will probably wonder why the filter doesn't work. So if this happens, check if you haven't made a new unfiltered query in the template.
```
Comment (lukasbestle) [inline content/1_docs/2_cookbook/3_templating/0_keep_your_code_dry/recipe.txt]:
```suggestion
If we now use the same template code as in the example above, we will probably be surprised why the filter doesn't work. So if this happens, check if you haven't made a new unfiltered query in the template.
```

File: content/1_docs/2_cookbook/3_templating/0_keep_your_code_dry/recipe.txt [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        <?php snippet('treemenu', ['subpages' => $p->children()]); ?>
+      </ul>
+    <?php endif ?>
+  </li>
+<?php endforeach ?>
+```
+
+## Takeaway
+
+Always store query results in variables when they are used multiple times. This improves your overall performance, keeps your code dry and easier to read, and prevents unexpected results.
```
Comment (lukasbestle) [inline content/1_docs/2_cookbook/3_templating/0_keep_your_code_dry/recipe.txt]:
```suggestion
Always store query results in variables when they are used multiple times. This improves your overall performance, keeps your code DRY and easier to read, and prevents unexpected results.
```

**Review by lukasbestle — COMMENTED** [review]
Awesome! A few minor comments:

**Pull request merged**